### DISCUSSION TITLE
open REPL in another tmux pane

### DISCUSSION THREAD

**Discussion by qrockz** [inducer/pudb#644] [General]
When hitting `!` the REPL console gets opened in the same pane.

1. Is it possible to open the REPL in another tmux pane while the TUI stays open?
2. And if so, can I do REPL command like continue and the TUI (code, variables, stack trace) get updated in real time?

**Comment by inducer** [comment]
I'd argue it doesn't matter whether it's a tmux pane or just any other terminal. Perhaps easiest way to offer this would be allow telnet'ing (or ssh'ing or whatever) into an already-running pudb. For this you would need event loop integration between urwid and something that can listen on sockets. [Urwid offers this](https://urwid.org/manual/mainloop.html#event-loops). Live updating the TUI base on this is within the realm of the possible, though you may have to do a full state scan to detect changes (which is fine... that's what pudb does every time user code finishes running).

**Reply by inducer** [reply]
PR welcome if you're so inclined.
